src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -2290,11 +2859,11 @@ ucs_status_t uct_ib_mlx5_devx_mem_attach(uct_md_h uct_md,
     ucs_strncpy_zero(access_key, uct_ib_mkey_token,
                      UCT_IB_MLX5DV_FLD_SZ_BYTES(alias_context, access_key));
 
-    memh->cross_mr = uct_ib_mlx5_devx_obj_create(md->super.dev.ibv_context, in,
-                                                 sizeof(in), out, sizeof(out),
-                                                 "MKEY_ALIAS",
-                                                 uct_md_attach_log_lvl(flags));
-    if (memh->cross_mr == NULL) {
+    umr_alias.cross_mr = uct_ib_mlx5_devx_obj_create(md->super.dev.ibv_context,
+                                                     in, sizeof(in), out,
+                                                     sizeof(out), "MKEY_ALIAS",
+                                                     uct_md_attach_log_lvl(flags));
+    if (umr_alias.cross_mr == NULL) {
```

Comment:
don't we need to set memh->cross_mr?

Reply:
Yes, we need to set it only when received mkey is NOT UMR mkey.
Thanks for spotting
Before my change `memh->cross_mr` field was used:
1. on the exporter side - to keep the created exported mkey, which is destroyed on `mem_unmap`
2. on the importer side - to keep the imported mkey alias, which is again destroyed on `mem_unmap`

But with UMR approach the situation is different on the importer side.
The whole idea was to keep the created aliases in the the hash map, so that their lifetimes are much longer than the lifetimes of memory handles. So with UMR approach we don't bind memh with alias anymore, and this is on purpose.
However we still need to support non-UMR mkeys, and this is what I missed here. When imported mkey is NOT UMR one, we still need to bind alias with memh, to make sure it gets destroyed on memh unmap:
```
    /* If received mkey is UMR key, store alias in the UMR hash map */
    if (uct_ib_mlx5_devx_umr_mkey_is_xgvmi(packed_mkey->lkey)) {
        umr_alias.lkey = memh->super.lkey;

        status = uct_ib_mlx5_devx_umr_mkey_hash_put(md, packed_mkey, &umr_alias);
        if (UCS_OK != status) {
            goto err_cross_mr_destroy;
        }
    } else {
        /* Not UMR mkey: move cross_mr ownership to memh */
        memh->cross_mr = umr_alias.cross_mr;
    }